platform/jewel/markdown/core/src/main/kotlin/org/jetbrains/jewel/markdown/processing/MarkdownProcessor.kt

```diff
@@ -186,8 +250,20 @@ public class MarkdownProcessor(
             }
 
             else -> null
+        }.also { block ->
+            if (scrollingSynchronizer != null && this is Block && block != null) {
+                postProcess(scrollingSynchronizer, this, block)
+            }
         }
 
+    private fun postProcess(scrollingSynchronizer: ScrollingSynchronizer, block: Block, mdBlock: MarkdownBlock) {
+        val spans =
+            block.sourceSpans.ifEmpty {
+                return
+            }
```

Comment:
This formatting looks a bit odd
```suggestion
        val spans = block.sourceSpans.takeIf { it.isNotEmpty() }
            ?: return
```